src/include/sof/common.h

```diff
@@ -14,8 +14,14 @@
 /* callers must check/use the return value */
 #define __must_check __attribute__((warn_unused_result))
 
+#ifdef __ZEPHYR__
+#include <zephyr/sys/util.h>
+#endif
+
 /* Align the number to the nearest alignment value */
+#ifndef IS_ALIGNED
```

Comment:
Except for "extreme" cases, `#ifndef` is always best avoided because it makes the usual, horrible macro experience even worse. Zephyr forbids it: https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#rule-a-3-macro-name-collisions

As requested in https://github.com/zephyrproject-rtos/zephyr/pull/67243#discussion_r1454373787, please rename this `SOF_ALIGN()` and make everything simpler: backwards and forwards compatibility and trannsitions

Reply:
> Can you do a follow-up on SOF side. 

Submitted in #8755